src/python/pants/build_graph/target.py

```diff
@@ -632,6 +638,40 @@ def traversable_dependency_specs(self):
     """
     return []
 
+  @classmethod
+  def compute_dependency_specs(cls, kwargs=None, payload=None):
+    """Given either pre-Target.__init__() kwargs or a post-Target.__init__() payload, compute the
+    full set of dependency specs in the same vein as the prior `traversable_dependency_specs`.
+
+    N.B. This is a temporary bridge to span the gap between v2 "Fields" products vs v1 `BuildGraph`
+    `Target` object representations. See:
+
+      https://github.com/pantsbuild/pants/issues/3560
+      https://github.com/pantsbuild/pants/issues/3561
+
+    :param dict kwargs: The pre-Target.__init__() kwargs dict.
+    :param Payload payload: The post-Target.__init__() Payload object.
+    :yields: Spec strings representing dependencies of this target.
```

Comment:
as it turns out, this is not possible without a `rel_path` context due to relative specs - e.g. `JvmTarget` depends on its `resources=` kwarg, which can contain specs like `':local-dep'`. so it seems string specs are ideal as they can be better parsed later with a `compute_dependency_specs`-external `rel_path` context.